File: android/app/src/main/java/com/firstvoice/app/data/model/ConversationSession.kt
package com.firstvoice.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ConversationSession(
    val id: String,
    val deviceId: String,
    val startedAt: Long,
    val endedAt: Long? = null,
    val gpsCoordinates: GPSCoordinate? = null,
    val responderLanguage: String,
    val survivorLanguage: String? = null,
    val interactions: List<Interaction> = emptyList(),
    val triageCardId: String? = null,
    val status: SessionStatus = SessionStatus.ACTIVE
)

@Serializable
enum class SessionStatus {
    ACTIVE, CLOSED, TIMED_OUT
}

@Serializable
sealed class Interaction {
    abstract val id: String
    abstract val timestamp: Long

    @Serializable
    data class SpeechTurn(
        override val id: String,
        override val timestamp: Long,
        val speaker: Speaker,
        val originalText: String,
        val originalLanguage: String,
        val translatedText: String,
        val translatedLanguage: String,
        val confidence: ConfidenceLevel
    ) : Interaction()

    @Serializable
    data class VisionAssessment(
        override val id: String,
        override val timestamp: Long,
        val photoId: String,
        val assessmentJson: String // Serialized DamageAssessment or InjuryAssessment
    ) : Interaction()

    @Serializable
    data class QuickPhraseUsage(
        override val id: String,
        override val timestamp: Long,
        val phraseId: String,
        val sourceText: String,
        val translatedText: String,
        val targetLanguage: String
    ) : Interaction()

    @Serializable
    data class Note(
        override val id: String,
        override val timestamp: Long,
        val text: String
    ) : Interaction()
}

@Serializable
enum class Speaker {
    RESPONDER, SURVIVOR
}

@Serializable
enum class ConfidenceLevel {
    HIGH, MEDIUM, LOW
}

File: packages/core/src/models/conversation-session.ts
import type {
  GPSCoordinate,
  ConfidenceLevel,
  DamageAssessment,
  InjuryAssessment,
} from './common';

/**
 * A time-bounded interaction between a responder and one or more survivors.
 * Contains an ordered sequence of speech turns, translations, vision assessments,
 * quick phrases, and notes. Produces a TriageCard when closed.
 */
export interface ConversationSession {
  /** UUID v4 unique identifier */
  id: string;
  /** Device that created this session */
  deviceId: string;
  /** Session start time (Unix epoch ms) */
  startedAt: number;
  /** Session end time, null if still active */
  endedAt: number | null;
  /** GPS location at session start */
  gpsCoordinates: GPSCoordinate | null;
  /** Responder's configured language */
  responderLanguage: string;
  /** Survivor's detected language, null until detected */
  survivorLanguage: string | null;
  /** Ordered log of all interactions */
  interactions: Interaction[];
  /** Linked triage card ID, set when session closes */
  triageCardId: string | null;
  /** Session lifecycle status */
  status: 'active' | 'closed' | 'timed_out';
}

/**
 * Union type for all interaction types within a session.
 */
export type Interaction =
  | SpeechTurnInteraction
  | VisionAssessmentInteraction
  | QuickPhraseInteraction
  | NoteInteraction;

export interface SpeechTurnInteraction {
  type: 'speech_turn';
  id: string;
  timestamp: number;
  speaker: 'responder' | 'survivor';
  originalText: string;
  originalLanguage: string;
  translatedText: string;
  translatedLanguage: string;
  confidence: ConfidenceLevel;
}

export interface VisionAssessmentInteraction {
  type: 'vision_assessment';
  id: string;
  timestamp: number;
  photoId: string;
  assessment: DamageAssessment | InjuryAssessment;
}

export interface QuickPhraseInteraction {
  type: 'quick_phrase';
  id: string;
  timestamp: number;
  phraseId: string;
  sourceText: string;
  translatedText: string;
  targetLanguage: string;
}

export interface NoteInteraction {
  type: 'note';
  id: string;
  timestamp: number;
  text: string;
}

export interface SessionFilter {
  status?: ('active' | 'closed' | 'timed_out')[];
  dateRange?: { start: number; end: number };
}
